plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PluginCompatTesterConfig.java

```diff
@@ -311,6 +317,18 @@ public void setMavenPropertiesFiles( String mavenPropertiesFile ) {
         return res;
     }
 
+    /**
+     * Gets Java version from the binary path to the <code>java</code> command.
+     * @return a string identifying the jvm in use
+     */
+    private String getJavaVersion(String javaCmdAbsolutePath) throws IOException {
```

Comment:
I have a concerne about this as the `String` produced by the build could potentially be very large and impact the performance of this.

Reply:
Can you be more specific? I do not think there's a concern here.

I indeed considered being a bit more defensive, but finally went for the simple solution expecting that java -fullversion should always work on most if not all common JVMs' `java` commands out there.

We _might_ see crashes if some `java` command returns garbage, but I fail to imagine some "performance" issue?

Reply:
:man_facepalming: the `Process` is not the entire build log of the plugin, but only a `java -fullVersion` command, so forget what I said. I'm still not a big fan of `String#replace` but the size of the `String` is way less than what I initially thought.